### PR TITLE
feat(OnyxTooltip): add property `open`

### PR DESCRIPTION
**Pull request by larsrickert**
Relates to #567

Add trigger property to trigger the tooltip on hover, click or boolean.

## Checklist

- [x] The added / edited code has been documented with [JSDoc](https://jsdoc.app/about-getting-started)
- [x] A changeset is added with `npx changeset add` if your changes should be released as npm package (because they affect the library usage)

### CODE REVIEW

File: packages/headless/src/composables/tooltip/createTooltip.ts [hunk 1]
```diff
@@ -1,19 +1,116 @@
...
 import { createBuilder } from "../../utils/builder";
 
-export const createTooltip = createBuilder(() => {
+export type CreateTooltipOptions = {
+  trigger: MaybeRef<TooltipTrigger | boolean>;
+  /**
+   * Number of milliseconds to use as debounce when showing/hiding the tooltip
+   * with openMode "hover".
+   *
+   * @default 0
```
Comment (JoCa96) [inline packages/headless/src/composables/tooltip/createTooltip.ts:11]: I would prefer an API that doesn't allow a delay to be set when open type is not "hover"

E.g.
```ts
type CreateTooltipOptions = {
    open: {
    type: 'hover';
    delay: number;
  } | "hover" | "click" | boolean;
}
```
or some using Generics

File: packages/headless/src/composables/tooltip/createTooltip.ts [hunk 2]
```diff
@@ -1,19 +1,116 @@
...
 import { createId } from "../..";
 import { createBuilder } from "../../utils/builder";
 
-export const createTooltip = createBuilder(() => {
+export type CreateTooltipOptions = {
+  trigger: MaybeRef<TooltipTrigger | boolean>;
+  /**
+   * Number of milliseconds to use as debounce when showing/hiding the tooltip
+   * with openMode "hover".
+   *
```
Comment (JoCa96) [inline packages/headless/src/composables/tooltip/createTooltip.ts:10]: A delay of `0` can be annoying, so I would prefer something like `1000`

File: packages/headless/src/composables/tooltip/createTooltip.ts [hunk 3]
```diff
@@ -1,19 +1,116 @@
...
     elements: {
-      trigger: {
+      trigger: computed(() => ({
         "aria-describedby": tooltipId,
-      },
-      tooltip: {
+        onClick: unref(options.trigger) === "click" ? handleClick : undefined,
+        ...hoverEvents.value,
+      })),
+      tooltip: computed(() => ({
```
Comment (JoCa96) [inline packages/headless/src/composables/tooltip/createTooltip.ts:106]: The tooltip content should not be able to receive focus, to be safe we should add `tabindex: -1`.
```suggestion
      tooltip: computed(() => ({
        tabindex: "-1",
```

File: packages/sit-onyx/src/components/OnyxTooltip/OnyxTooltip.vue [hunk 4]
```diff
@@ -18,11 +20,15 @@ defineSlots<{
...
+  state: { isVisible },
+} = createTooltip({
+  trigger: computed(() => props.trigger),
+  debounce: 200,
+});
 </script>
 
 <template>
-  <div class="onyx-tooltip-wrapper">
+  <div class="onyx-tooltip-wrapper" :class="{ 'onyx-tooltip-wrapper--hidden': !isVisible }">
```
Comment (JoCa96) [inline packages/sit-onyx/src/components/OnyxTooltip/OnyxTooltip.vue:31]: Might be nicer to set the class on the tooltip element, where it is actually applied.

**Review by JoCa96 — CHANGES_REQUESTED** [review]

File: packages/headless/src/composables/tooltip/createTooltip.ts [hunk 5]
```diff
@@ -1,19 +1,116 @@
...
-export const createTooltip = createBuilder(() => {
+export type CreateTooltipOptions = {
+  trigger: MaybeRef<TooltipTrigger | boolean>;
+  /**
+   * Number of milliseconds to use as debounce when showing/hiding the tooltip
+   * with openMode "hover".
+   *
+   * @default 0
+   */
+  debounce?: number;
```
[the remark this answers is not in the document]
Comment (larsrickert) [reply packages/headless/src/composables/tooltip/createTooltip.ts:13]: - delay: if multiple events are fired, of all them get executed
- debounce: if multiple events are fired, only the last one gets executed

So we currently have (and need) a debounce, not a delay :)

File: packages/headless/src/composables/tooltip/createTooltip.ts (hunk 2 above)
Comment (larsrickert) [reply packages/headless/src/composables/tooltip/createTooltip.ts:10]: I set a default value of 200 which can be overridden if needed

File: packages/headless/src/composables/tooltip/createTooltip.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (BoppLi) [reply packages/headless/src/composables/tooltip/createTooltip.ts]: from our 2022 sitcon talk, if helpful:
- throttle: Reduces trigger rate - recommended for scroll/resize events
- debounce: Waits until firing stops - recommended for input events

**Review by JoCa96 — APPROVED** [review]

**Pull request merged**